utils/hct/hctdb.py

```diff
@@ -5509,9 +5512,38 @@ def UFI(name, **mappings):
         next_op_idx += 1
 
         # End of DXIL 1.8 opcodes.
+
+        # RayQuery
+        self.add_dxil_op(
+            "AllocateRayQuery2",
+            next_op_idx,
+            "AllocateRayQuery2",
+            "allocates space for RayQuery and return handle",
+            "v",
+            "",
+            [
+                db_dxil_param(0, "i32", "", "handle to RayQuery state"),
+                db_dxil_param(
+                    2,
+                    "u32",
+                    "constRayFlags",
+                    "Valid combination of RAY_FLAGS",
+                    is_const=True,
+                ),
+                db_dxil_param(
+                    3,
+                    "u32",
+                    "constRayQueryFlags",
+                    "Valid combination of RAYQUERY_FLAGS",
+                    is_const=True,
+                ),
+            ],
+        )
+        next_op_idx += 1
+
         self.set_op_count_for_version(1, 8, next_op_idx)
-        assert next_op_idx == 258, (
-            "258 is expected next operation index but encountered %d and thus opcodes are broken"
+        assert next_op_idx == 259, (
+            "259 is expected next operation index but encountered %d and thus opcodes are broken"
```

Comment:
I think you misunderstood my previous comment. This change should not be made.  This block of code should remain with the comment `End of DXIL 1.8 opcodes.`, and the new opcode should appear below this assert.

Otherwise, the new opcode is considered part of DXIL 1.8.  The point of this block of code between each DXIL version is to prevent changing opcodes for an already-shipped DXIL version.